### PR TITLE
Fixed #29598 -- Deprecated FloatRangeField in favor of DecimalRangeField.

### PR DESCRIPTION
**Pull request by StefanoChiodino**
https://code.djangoproject.com/ticket/29598

### CODE REVIEW

File: django/contrib/postgres/fields/ranges.py [hunk 1]
```diff
@@ -100,15 +101,21 @@ def db_type(self, connection):
         return 'int8range'
 
 
-class FloatRangeField(RangeField):
+class DecimalRangeField(RangeField):
     base_field = models.FloatField
```
Comment (charettes) [inline django/contrib/postgres/fields/ranges.py]: This should be `models.DecimalField`?

File: django/contrib/postgres/fields/ranges.py [hunk 2]
```diff
@@ -100,15 +101,21 @@ def db_type(self, connection):
...
-    form_field = forms.FloatRangeField
+    form_field = forms.DecimalRangeField
 
     def db_type(self, connection):
         return 'numrange'
 
 
+class FloatRangeField(DecimalRangeField):
+    def __init__(self, *args, **kwargs):
+        warnings.warn('FloatRangeField has been renamed to DecimalRangeField and is now deprecated')
```
Comment (charettes) [inline django/contrib/postgres/fields/ranges.py]: `stacklevel=2` to have the warning show at `FloatRangeField` initialization time?

File: django/contrib/postgres/forms/ranges.py [hunk 3]
```diff
@@ -66,12 +68,17 @@ class IntegerRangeField(BaseRangeField):
...
 
-class FloatRangeField(BaseRangeField):
+class DecimalRangeField(BaseRangeField):
     default_error_messages = {'invalid': _('Enter two numbers.')}
     base_field = forms.FloatField
     range_type = NumericRange
 
 
+class FloatRangeField(DecimalRangeField):
+    warnings.warn('FloatRangeField has been renamed to DecimalRangeField and is now deprecated')
```
Comment (charettes) [inline django/contrib/postgres/forms/ranges.py]: This needs to be done at `__init__` as well, right now it happens at import time.

File: django/contrib/postgres/forms/ranges.py [hunk 4]
```diff
@@ -66,12 +68,17 @@ class IntegerRangeField(BaseRangeField):
     range_type = NumericRange
 
 
-class FloatRangeField(BaseRangeField):
+class DecimalRangeField(BaseRangeField):
     default_error_messages = {'invalid': _('Enter two numbers.')}
     base_field = forms.FloatField
```
Comment (charettes) [inline django/contrib/postgres/forms/ranges.py]: `forms.DecimalField`

File: docs/ref/contrib/postgres/fields.txt [hunk 5]
```diff
@@ -658,6 +658,15 @@ excluded; that is, ``[)``.
...
+
+        Renamed to ``DecimalRangeField``.
+
+``DecimalRangeField``
+---------------------
+
+.. class:: DecimalRangeField(**options)
+
     Stores a range of floating point values. Based on a
     :class:`~django.db.models.FloatField`. Represented by a ``numrange`` in the
```
Comment (Ian-Foote) [inline docs/ref/contrib/postgres/fields.txt]: This needs to be updated to `django.db.models.DecimalField`.

**Comment by timgraham** [comment]
We use [system checks](https://docs.djangoproject.com/en/dev/topics/migrations/#considerations-when-removing-model-fields) for model field deprecations.

**Pull request merged**